src/Classes/ModStore.lua

```diff
@@ -501,8 +501,38 @@ function ModStoreClass:EvalMod(mod, cfg)
 				return
 			end
 		elseif tag.type == "SocketedIn" then
-			if not cfg or tag.slotName ~= cfg.slotName or (tag.keyword and (not cfg or not cfg.skillGem or not calcLib.gemIsType(cfg.skillGem, tag.keyword))) then
+			if not cfg or tag.slotName ~= cfg.slotName or (tag.socketColor ~= nil and (tag.sockets == nil or (tag.sockets ~= nil and type(tag.sockets) ~= "string")) and tag.socketColor ~= cfg.socketColor) or (tag.keyword and (not cfg or not cfg.skillGem or not calcLib.gemIsType(cfg.skillGem, tag.keyword))) then
 				return
+			elseif tag.sockets ~= nil then
+				if tag.sockets == "all" and tag.socketColor ~= nil then
+					local targetAtrColor = tag.socketColor == "R" and "strengthGems" or tag.socketColor == "G" and "dexterityGems" or tag.socketColor == "B" and "intelligenceGems"
+					local total = (cfg.intelligenceGems or 0) + (cfg.dexterityGems or 0) + (cfg.strengthGems or 0)
+					local count = cfg[targetAtrColor] or 0
+					if total == 0 or count == 0 or total > count then
+						return
```

Comment:
This could be changed to allow the tag.neg to work on some sections as i believe it could be useful in the future to have not all ___ gems and not a green gem support instead of just not a green gem in sockets {1,2,3,4,5}
Maybe changing
elseif tag.sockets ~= nil then -> elseif tag.sockets ~= nil and tag.socketColor ~= nil then
which lets you drop socketColour from the other if statements and using a 
local match = true 
at the start of the inner set of if statements and where the returns are setting
match = false
then flipping it after the if statements like you currently do with notFound and deciding whether to return.
Similar to other implementations in this function.

Reply:
Pulling out `tag.socketColor ~= nil` would require refactoring to make all mods with with 
colors instead of tags. It may not be a bad idea to do but then i feel like `socketColor` should be renamed to something else as i wanted to distinguish between socket colors and gem color by using keywords for gems and socketColor for sockets. Maybe just rename it to color ?
```LUA
elseif tag.sockets ~= nil then
    local match = false
    if tag.sockets == "all" and tag.socketColor ~= nil then
        local targetAtrColor = tag.socketColor == "R" and "strengthGems" or tag.socketColor == "G" and "dexterityGems" or tag.socketColor == "B" and "intelligenceGems"
        local total = (cfg.intelligenceGems or 0) + (cfg.dexterityGems or 0) + (cfg.strengthGems or 0)
        local count = cfg[targetAtrColor] or 0
        if total == 0 or count == 0 or total > count then
            return
        end
        match = true
    elseif type(tag.sockets) == "number" and tag.socketColor ~= nil then
        local targetAtrColor = tag.socketColor == "R" and "strengthGems" or tag.socketColor == "G" and "dexterityGems" or tag.socketColor == "B" and "intelligenceGems"
        local count = cfg[targetAtrColor] or 0
        if count == 0 or count < tag.sockets then
           return
        end
        match = true
    elseif type(tag.sockets) == "table" then
        if cfg.socketNum == nil then return end
            for _, val in ipairs(tag.sockets) do
                if val == cfg.socketNum then
                    match = true
                    break
                end
            end
        end
        if tag.neg then
            match = not match
        end
        if not match then
            return
        end
    end
```

Reply:
I think the way this code works is you need to return on match = false. You can probably return on some kinda of error regardless.
This means all returns that aren't "invalid" should be replaced with match = false and if it kinda assumes a match otherwise then match = true near the start of the if statement.

Reply:
I think i understand what you mean now. Should be solved by https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/4981/commits/569b6358c826c481578e7e030ae482e22feb93d2